### ISSUE TITLE
Support negative padding sizes with `torch.nn.functional.pad`

### ISSUE THREAD

**Issue by avery-blanchard** [torch-spyre/torch-spyre#1393]
### What feature would you like to be added?

The support for`torch.nn.functional.pad` added in #1310 should be extended to support negative padding. 

### Why is this feature needed?

The `torch.nn.functional.pad` operation supports both adding and removing padding.

**Comment by michihirohorie** [comment]
Following is my understanding and open questions. Please correct me if anything is wrong.

- Negative padding can be handled by the slice op 
Negative padding corresponds to pruning the input tensor and can be implemented as a slicing op.

- Apply slice then overwrite 
Slice op defines new iteration space by removing unused regions. After this iteration space is finalized we can apply overwrite.

- Slice view can be passed directly to overwrite
No need of explicit copy before the overwrite

- A corner case - resulting in a zero-sized dim after slicing
If slicing results in a tensor with size 0 in any dimension, how should overwrite behave?
